javascript/ql/src/meta/MetaMetrics.qll

```diff
@@ -0,0 +1,32 @@
+/**
+ * Helpers to generating meta metrics, that is, metrics about the CodeQL analysis and extractor.
+ */
+
+private import javascript
+private import semmle.javascript.dependencies.Dependencies
+private import semmle.javascript.dependencies.FrameworkLibraries
+private import semmle.javascript.frameworks.Testing
+
+/**
+ * Gets the root folder of the snapshot.
+ *
+ * This is selected as the location for project-wide metrics.
+ */
+Folder projectRoot() { result.getRelativePath() = "" }
+
+/** A file we ignore because it is a test file or compiled/generated/bundled code. */
+class IgnoredFile extends File {
+  IgnoredFile() {
```

Comment:
How about using `ClassifyFiles::classify/2` here?
(or in a different PR, since this PR just moves the class)

Reply:
My worry was that changes to classification would make metrics incomparable over time so I'd rather have something that excludes the most obviously uninteresting code but otherwise is fixed.

This was something that bit me when I tried to classify external calls in order measure the relative call graph completeness more accurately. It meant the goal posts kept moving and the results were much harder to interpret.

But now is definitely a good time to review this class so please suggest improvements if you think of any.

Reply:
I see. Lets keep it simple then. 

Perhaps we should measure the size of `classify/2` as a metric as well. If we see that it is stable in practice, then there's little harm in focusing our results with it.

(Hmm. We will even have problems without `classify/2` moving goal posts. `js/meta/tainted-nodes` will decrease when `js/meta/sanitizers-reachable-from-source` increases...)

Reply:
If `js/meta/tainted-nodes` decreases due to the addition of more sanitizers, isn't that just the case of some of the old ones being FPs? All the metrics can decrease when removing FPs.

Reply:
Right. Never mind that latter comment.